### DISCUSSION TITLE
fulfillment asks for approval from seaport contract instead of the conduit

### DISCUSSION THREAD

**Discussion by ryespresso** [ProjectOpenSea/opensea-js#1248] [Q&A] [answered]
### Component

Utils

### Have you ensured that all of these are up to date?

- [X] opensea-js
- [X] Node (minimum v16)

### What version of opensea-js are you on?

2.0.7

### What function is the bug in?

fullfillOrder

### Operating System

macOS (Apple Silicon)

### Describe the bug

When using `createOffer` function to create a listing, the user is asked to approve the conduit to move the NFT. This is the desired outcome.

However, if someone creates a bid on that NFT and I choose to fulfill that bid (as seen below) I'm asked to approve the seaport contract to move the NFT and not the conduit. Shouldn't it be the conduit?
```
const { executeAllActions: executeAllFulfillActions } = await seaport.fulfillOrder({
     order: { parameters, signature },
     accountAddress,
});

const transaction = await executeAllFulfillActions();
```

I should note that the parameters of the fulfillment include: `"conduitKey":"0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",`

**Comment by ryanio** [comment]
Are you sure it's asking for approval for the NFT on accept offer and not WETH? This is because fulfillOrder will make your account transfer WETH from seaport to your account. To get around this, you can build a counter order and use matchOrders, this is how we do it on OpenSea if you want to try accepting an offer on testnets.opensea.io

It also depends on the full order if it's 721 or 1155 full or partial, if you want to share the whole order.

**Reply by ryespresso** [reply]
Yes @ryanio - it is asking for the approval for the NFT because I am accepting the bid to send my NFT away. The bidder has already done their own WETH approvals when creating the bid.

The bug is that it is asking me to approve seaport contract and not the conduit. Here is the full order that is being fulfilled:

```
{
  "salt":"0x000000000000000000000000000000000000000000000000926f38c0ad6c64a3",
  "zone":"0x0000000000000000000000000000000000000000",
  "offer":[
    {
      "token":"0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6",
      "itemType":1,
      "endAmount":"2000000000000000",
      "startAmount":"2000000000000000",
      "identifierOrCriteria":"0"
    }
  ],
  "counter":"0",
  "endTime":"115792089237316195423570985008687907853269984665640564039457584007913129639935",
  "offerer":"0x14d86d7354b794c4b38d4dbfd525eeeef9e910ab",
  "zoneHash":"0x0000000000000000000000000000000000000000000000000000000000000000",
  "orderType":0,
  "startTime":"1697810481",
  "conduitKey":"0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000",
  "consideration":[
    {
      "token":"0xD6a39123Ce772799D6A6cBeEAC7F5490c94f7F71",
      "itemType":2,
      "endAmount":"1",
      "recipient":"0x14d86d7354b794c4b38d4dbfd525eeeef9e910ab",
      "startAmount":"1",
      "identifierOrCriteria":"72859713366585270946722344216477908029499397131298552184592514533884178151937"
    },
    {
      "token":"0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6",
      "itemType":1,
      "endAmount":"20000000000000",
      "recipient":"0xB69319B9B3Eb6cD99f5379b9b3909570F099652a",
      "startAmount":"20000000000000",
      "identifierOrCriteria":"0"
    }
  ],
  "totalOriginalConsiderationItems":2
}
```

**Reply by ryanio** [reply, marked as answer]
you made this discussion in opensea-js but you are using seaport-js. if you were to use opensea-js i believe we provide the opensea conduitKey automatically in fulfillment. in seaport-js you have to initialize seaport with overrides.defaultConduitKey or provide conduitKey in fulfillOrder to pass the fulfiller conduitKey.

**Reply by ryespresso** [reply]
Ah, I apologize; wrong `*-js` library. Thank you, this actually did the trick. Odd that it doesn't look at the conduit key in the actual order it's fulfilling. Feel free to close.

**Reply by ryanio** [reply]
great! yes there is a difference between the offerer's conduit key and the fulfiller's as they can be different values. thanks!
